httpd/http_server.c

```diff
@@ -0,0 +1,333 @@
+    // "//api/v1/info" to be caled instead of "/api/v1/info"
+    if (norm_path.len > 2 &&
+        norm_path.base[0] == '/' &&
+        norm_path.base[1] == '/' ) {
+            norm_path.base++;
+            norm_path.len--;
+    }
+
+    size_t api_loc = h2o_strstr(norm_path.base, norm_path.len, H2O_STRLIT(API_V1_PREFIX));
+    if (api_loc == SIZE_MAX)
+        return 1;
+
+    h2o_iovec_t api_command = norm_path;
+    api_command.base += api_loc + strlen(API_V1_PREFIX);
+    api_command.len -= api_loc + strlen(API_V1_PREFIX);
+
+    if (!api_command.len)
+        return 1;
+
+    // this (emulating struct web_client) is a hack and will be removed
+    // in future PRs but needs bigger changes in old http_api_v1
+    // we need to make the web_client_api_request_v1 to be web server
+    // agnostic and remove the old webservers dependency creep into the
+    // individual response generators and thus remove the need to "emulate"
+    // the old webserver calling this function here and in ACLK
+    struct web_client w;
+    w.response.data = buffer_create(NBUF_INITIAL_SIZE_RESP, NULL);
+    w.response.header = buffer_create(NBUF_INITIAL_SIZE_RESP, NULL);
+    w.decoded_query_string[0] = 0;
+    w.acl = WEB_CLIENT_ACL_DASHBOARD;
+
+    char *path_c_str = iovec_to_cstr(&api_command);
+    char *path_unescaped = url_unescape(path_c_str);
+    freez(path_c_str);
+
+    IF_HAS_URL_PARAMS(req) {
+        h2o_iovec_t query_params = URL_PARAMS_IOVEC_INIT_WITH_QUESTIONMARK(req);
+        char *query_c_str = iovec_to_cstr(&query_params);
+        char *query_unescaped = url_unescape(query_c_str);
+        freez(query_c_str);
+        strcpy(w.decoded_query_string, query_unescaped);
+        freez(query_unescaped);
+    }
+
+    web_client_api_request_v1(*host, &w, path_unescaped);
+    freez(path_unescaped);
+
+    h2o_iovec_t body = buffer_to_h2o_iovec(w.response.data);
+
+    // we move msg body to req->pool managed memory as it has to
+    // live until whole response has been encrypted and sent
+    // when req is finished memory will be freed with the pool
+    void *managed = h2o_mem_alloc_shared(&req->pool, body.len, NULL);
+    memcpy(managed, body.base, body.len);
+    body.base = managed;
+
+    req->res.status = 200;
+    req->res.reason = "OK";
+    if (w.response.data->content_type == CT_APPLICATION_JSON)
+        h2o_add_header(&req->pool, &req->res.headers, H2O_TOKEN_CONTENT_TYPE, NULL, CONTENT_JSON_UTF8);
+    else
+        h2o_add_header(&req->pool, &req->res.headers, H2O_TOKEN_CONTENT_TYPE, NULL, CONTENT_TEXT_UTF8);
+    h2o_start_response(req, &generator);
+    h2o_send(req, &body, 1, H2O_SEND_STATE_FINAL);
+
+    buffer_free(w.response.data);
+    buffer_free(w.response.header);
+
+    return 0;
+}
+
+static int netdata_uberhandler(h2o_handler_t *self, h2o_req_t *req)
+{
+    RRDHOST *host = localhost;
+
+    int ret = _netdata_uberhandler(self, req, &host);
+
+    char host_uuid_str[GUID_LEN + 1];
```

Comment:
Considering that we are working here with `UUID`, I suggest to use [UUID_STR_LEN](https://github.com/netdata/netdata/blob/513452aa447759acecb16504673db7dce40ffaa2/libnetdata/libnetdata.h#L114-L118) instead `GUID_LEN + 1`.